src/cli/commands/lint.ts

```diff
@@ -69,33 +78,32 @@ linting ./openapi.yaml
   }
 }
 
-async function lint(name: string, flags: any, command: Lint) {
-  command.log(`linting ${name}`);
-  let obj: IParserResult;
-  try {
-    obj = await readInputArguments(name, flags.encoding);
-  } catch (ex) {
-    throw new Error(`Could not parse ${name}: ${ex.message}`);
-  }
+async function lint(name: string, flags: any, command: Lint, customRuleset?: RuleCollection) {
+  command.log(`Linting ${name}`);
+  const spec: IParserResult = await readParsable(name, flags.encoding);
 
   const spectral = new Spectral();
-  if (obj.data.swagger && obj.data.swagger === '2.0') {
+  if (parseInt(spec.data.swagger) === 2) {
     command.log('OpenAPI 2.0 (Swagger) detected');
     spectral.addFunctions(oas2Functions());
     spectral.addRules(oas2Rules());
-  } else if (obj.data.openapi && typeof obj.data.openapi === 'string' && obj.data.openapi.startsWith('3.')) {
+  } else if (parseInt(spec.data.openapi) === 3) {
     command.log('OpenAPI 3.x detected');
     spectral.addFunctions(oas3Functions());
     spectral.addRules(oas3Rules());
   } else {
     throw new Error('Input document specification type could not be determined');
   }
 
+  if (customRuleset) {
```

Comment:
Can we think of ruleset a more like “did they tell us what ruleset to use? No? Fine, we’ll take a guess”? Right now this is doing both. Remember if they provide a ruleset we don’t want to add oas2/3 automatically (they can require that later).

Reply:
Yup, I remember, I wanted to do it in next PR though. Sounds good?

Reply:
Thew require is slated to be done later absolutely, but I am talking about the fact that right now it is autoloading oas2/oas3 and then also `--ruleset=foo`. 

We should not add a feature to master and then change it later. The precedence is:
1. Load the rulesets if provided
2. Else autodetect

You just need to move the autodetect a few lines down.

Reply:
Coolio, yup 
> We should not add a feature to master and then change it later. 
you bought me with this argument. Will fix.

Reply:
done